src/MediawikiApi.php

```diff
@@ -351,19 +351,52 @@ private function getUserAgent() {
 	 * @param $result
 	 */
 	private function logWarnings( $result ) {
-		if ( is_array( $result ) && array_key_exists( 'warnings', $result ) ) {
-			foreach ( $result['warnings'] as $module => $warningData ) {
-				// Accomodate both formatversion=2 and old-style API results
-				$logPrefix = $module . ': ';
-				if ( isset( $warningData['*'] ) ) {
-					$this->logger->warning( $logPrefix . $warningData['*'], [ 'data' => $warningData ] );
-				} else {
-					$this->logger->warning( $logPrefix . $warningData['warnings'], [ 'data' => $warningData ] );
+		if ( is_array( $result ) ) {
+			// Let's see if there is 'warnings' key on the first level of the array...
+			if ( $this->logWarning( $result ) ) {
+				return;
+			}
+
+			// ...if no then go one level deeper and check there for it.
+			foreach ( $result as $value ) {
+				if ( !is_array( $value ) ) {
+					continue;
 				}
+
+				$this->logWarning( $value );
 			}
 		}
 	}
 
+	/**
+	 * @param $array
+	 *
+	 * @return bool Whether any warning has been logged or not.
+	 */
+	private function logWarning( $array ) {
```

Comment:
This should be proteccted, in case subclasses want to do something different with it.